component/azstorage/file_share.go

```diff
@@ -150,86 +154,259 @@ func (fs *FileShare) TestPipeline() error {
 }
 
 func (fs *FileShare) ListContainers() ([]string, error) {
-	return nil, nil
+	log.Trace("FileShare::ListContainers : Listing containers")
+	cntList := make([]string, 0)
+
+	marker := azfile.Marker{}
+	for marker.NotDone() {
+		resp, err := fs.Service.ListSharesSegment(context.Background(), marker, azfile.ListSharesOptions{})
+		if err != nil {
+			log.Err("FileShare::ListContainers : Failed to get container list")
+			return cntList, err
+		}
+
+		for _, v := range resp.ShareItems {
+			cntList = append(cntList, v.Name)
+		}
+
+		marker = resp.NextMarker
+	}
+
+	return cntList, nil
 }
 
 // This is just for test, shall not be used otherwise
 func (fs *FileShare) SetPrefixPath(string) error {
-	return nil
+	return syscall.ENOTSUP
 }
 
 func (fs *FileShare) Exists(name string) bool {
-	return false
+	log.Trace("FileShare::Exists : name %s", name)
+	if _, err := fs.GetAttr(name); err == syscall.ENOENT {
+		return false
+	}
+	return true
 }
+
 func (fs *FileShare) CreateFile(name string, mode os.FileMode) error {
-	return nil
+	return syscall.ENOTSUP
 }
 func (fs *FileShare) CreateDirectory(name string) error {
-	return nil
+	return syscall.ENOTSUP
 }
 func (fs *FileShare) CreateLink(source string, target string) error {
-	return nil
+	return syscall.ENOTSUP
 }
 
 func (fs *FileShare) DeleteFile(name string) error {
-	return nil
+	return syscall.ENOTSUP
 }
 func (fs *FileShare) DeleteDirectory(name string) error {
-	return nil
+	return syscall.ENOTSUP
 }
 
 func (fs *FileShare) RenameFile(string, string) error {
-	return nil
+	return syscall.ENOTSUP
 }
 func (fs *FileShare) RenameDirectory(string, string) error {
-	return nil
+	return syscall.ENOTSUP
 }
 
+// GetAttr : Retrieve attributes of a file or directory
 func (fs *FileShare) GetAttr(name string) (attr *internal.ObjAttr, err error) {
-	return nil, nil
+	log.Trace("FileShare::GetAttr : name %s", name)
+
+	path := filepath.Join(fs.Config.prefixPath, name)
+	splitPath := strings.Split(path, "/")
+	splitPath = splitPath[:len(splitPath)-1]
+	joinedPath := strings.Join(splitPath, "/")
+
+	fileURL := fs.Share.NewDirectoryURL(joinedPath).NewFileURL(filepath.Base(name))
+	log.Trace("FileShare::GetAttr : getting fileeeeeeeeeeee properties for %s, %s, %s", fs.Config.prefixPath, name, fileURL)
+	prop, err := fileURL.GetProperties(context.Background())
+
+	if err == nil { // file
+		attr = &internal.ObjAttr{
+			Path:   name, // We don't need to strip the prefixPath here since we pass the input name
+			Name:   filepath.Base(name),
+			Size:   prop.ContentLength(),
+			Mode:   0,
+			Mtime:  prop.LastModified(),
```

Comment:
same for Directory Properties

Reply:
the file/directory response has methods for ChangeTime and CreationTime but their return value is of type string rather than time.Time (seen in zz_generated_models.go); am i looking in the wrong place though?

Reply:
maybe try to see if you can look into how to convert a string to a time.time